### PR TITLE
Dnd content

### PR DESCRIPTION
**Pull request by neron2150**

### CODE REVIEW

File: src/dnd/DragArea.jsx [hunk 1]
```diff
@@ -112,24 +119,65 @@ class DragArea extends Component {
...
+        draggables:
+        {
+          ...newDroppable.draggables,
+          [draggableID]: {
+            ID: draggableID,
+            containerID: newDroppableID,
+          },
+        },
+      };
+      delete containers[lastDroppableID].draggables[draggableID];
```
Comment (olyverDev) [inline src/dnd/DragArea.jsx]: Get rid of this

Maybe add set / unset utils for objects

File: src/dnd/Draggable.jsx [hunk 2]
```diff
@@ -1,31 +1,27 @@
...
-          >
-            <p>{this.props.id}</p>
-            {this.props.children}
-          </div>
-        )}
-      </DndContext.Consumer>
+
+      <div // eslint-disable-line jsx-a11y/no-static-element-interactions
+        className="draggable"
+        onMouseDown={this.mouseDown(this.props.setDraggable)}
```
Comment (olyverDev) [inline src/dnd/Draggable.jsx]: No sense to pass `this.props.setDraggable`

It's inside props

Restructure `mouseDown`

File: src/dnd/DragArea.jsx [hunk 3]
```diff
@@ -112,24 +119,65 @@ class DragArea extends Component {
...
+        ID={container.ID}
+        setDroppable={this.setDroppable}
+      >
+        {this.props.renderDroppableByID(
+          draggables, container)}
+      </DroppableContainer>
+    );
+  };
+
+  renderContent = () => Object.values(this.state.containers).map(this.renderDroppableContainer);
```
Comment (olyverDev) [inline src/dnd/DragArea.jsx]: Redundant

Just use this code inside render

File: src/components/Board.jsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import React, { Component } from 'react';
+import './App.css';
+import DragArea from '../dnd/DragArea';
+
+const CREATE_TASKS = (count, containerID) => {
```
Comment (olyverDev) [inline src/components/Board.jsx]: Move out

File: src/components/Board.jsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  const tasks = {};
+  for (let i = 0; i <= count; i++) {
+    const id = Math.random().toString(36).substring(7);
+    tasks[id] = { ID: id, containerID };
+  }
+
+  return tasks;
+};
+
+const createContent = () => {
```
Comment (olyverDev) [inline src/components/Board.jsx]: Move out and all mock data below

Just import what's needed

File: src/components/Board.jsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  // };
+
+  render() {
+    return (
+      <DragArea
+        onDrop={this.onDrop}
+        containers={createContent()}
+        renderDroppableByID={this.renderDroppableByID}
+        renderDraggableByID={this.renderDraggableByID}
+        shouldRebase
```
Comment (olyverDev) [inline src/components/Board.jsx:30]: Strange, what does it mean ?

Is it used ?

File: src/dnd/DragArea.jsx [hunk 7]
```diff
@@ -10,13 +11,15 @@ class DragArea extends Component {
     lastY: 0,
     deg: 0,
     draggableId: null,
-    droppableId: null,
+    lastDroppableId: null,
     droppableContainers: [],
+    containers: this.props.containers,
```
Comment (olyverDev) [inline src/dnd/DragArea.jsx:16]: Remember it's setting only on initialization

If u expect change of this props you should implement `getDerivedStateFromProps` lifecycle

Do u expect it?

File: src/dnd/DragArea.jsx [hunk 8]
```diff
@@ -10,13 +11,15 @@ class DragArea extends Component {
...
   };
 
   draggable = null;
 
   setDraggable = (draggable, draggableId) => {
+    // TODO: resolve the problem with finding z-index
```
Comment (olyverDev) [inline src/dnd/DragArea.jsx:22]: Let's resolve right now with help of not used `classnames`

This PR should be concluding regarding base dnd UI

File: src/dnd/DragArea.jsx [hunk 9]
```diff
@@ -60,30 +64,33 @@ class DragArea extends Component {
...
+      lastX: e.pageX,
+      lastY: e.pageY,
     }));
   };
 
-  onDrop = (draggableId, droppableId) => {
-    this.props.onDrop(draggableId, droppableId);
+  onDrop = (draggableId, newDroppableId, lastDroppableId) => {
+    if (this.props.onDrop) this.props.onDrop(draggableId, newDroppableId, lastDroppableId);
+    if (this.props.shouldRebase) this.rebaseDraggable(draggableId, newDroppableId, lastDroppableId);
```
Comment (olyverDev) [inline src/dnd/DragArea.jsx:93]: Strange check

If I pass `false` drop won't happen or what ?

File: src/components/Board.jsx (hunk 4 above)
Comment (neron2150) [reply src/components/Board.jsx]: moved to mocks/BoardMocks.js

File: src/components/Board.jsx (hunk 5 above)
Comment (neron2150) [reply src/components/Board.jsx]: moved to mocks/BoardMocks.js

File: src/dnd/DragArea.jsx (hunk 3 above)
Comment (neron2150) [reply src/dnd/DragArea.jsx]: Ready

File: src/dnd/Draggable.jsx (hunk 2 above)
Comment (neron2150) [reply src/dnd/Draggable.jsx]: ready

**Pull request merged**